static/js/pages/applications/ApplicationDashboardPage.js

```diff
@@ -72,57 +78,277 @@ export class ApplicationDashboardPage extends React.Component<Props, State> {
     this.onCollapseToggle(applicationId)
   }
 
-  renderApplicationDetail = (applicationId: number) => {
-    const {
-      allApplicationDetail,
-      setDrawerOpen,
-      setDrawerState,
-      setDrawerMeta
-    } = this.props
+  /*
+   * Returns an object mapping an application step id to the user's submission (if anything was submitted
+   * for that step).
+   */
+  getAppStepSubmissions = (
```

Comment:
It would be good to refactor this out to a library/util method since it doesn't access anything on the class.